### DISCUSSION TITLE
Instruction format

### DISCUSSION THREAD

**Discussion by spqw** [ggml-org/llama.cpp#3164] [Q&A]
When running llama.cpp server with a GGUF model, is it needed to specify the instruction format?
If so, can someone explain me how to do so?
For example, for the model [Phind-CodeLlama-34B-v2](https://huggingface.co/Phind/Phind-CodeLlama-34B-v2), the following instruction format is suggested:
```
### System Prompt
You are an intelligent programming assistant.

### User Message
Implement a linked list in C++

### Assistant
...
```
Where can I specify this?

NB: I'm not talking about specifying an output format or any grammar. I'm talking about formatting the prompt so that it fits what the model expects.

**Comment by staviq** [comment]
Copy paste that suggested format into a text file, remove the "..." at the end, replace "Implement a linked list in C++" with your question, and use that as a prompt.

You can tell `main` to read prompt from a file with `-f filename`.

That's all for a single run mode.

If you want to use interactive mode, asking more than one question in a single "session", frankly, it's easier to use `server` for that, but you can do that with `main` too, here is the readme section for it: https://github.com/ggerganov/llama.cpp#interactive-mode and you get a more detailed description of parameters by running `./main -h`, and take a look at the scripts in `examples` directory.
